packages/plugin-hardhat/src/upgrade-proxy.ts

```diff
@@ -16,27 +16,32 @@ import {
 
 import { getProxyAdminFactory } from './proxy-factory';
 import { readValidations } from './validations';
-import { deploy } from './utils/deploy';
+import { defaultDeploy, DeploymentExecutor, intoCoreDeployment } from './utils/deploy';
+
+export interface UpgradeOptions extends ValidationOptions {
+  executor?: DeploymentExecutor;
```

Comment:
I think this should be a "tx executor" and not only a deployment executor.

I'm thinking that for `upgradeProxy` it should also take care of sending the `upgrade` function call. What do you think?

Reply:
Seems legit, I will look into this!